README.md

```diff
@@ -1,116 +1,208 @@
+    * If both contributions deleted or renamed with body changes, we immediately report a conflict if they renamed to different signatures. Otherwise, we merge their contents using textual merge.
+    * If one renamed without body changes and the other deleted renamed with body changes, we report a conflict if they renamed to different signatures or if they renamed to equal signatures but the latter included a reference to the renamed method/constructor.
 
-* Merging 3 directories:
+### [New Element Referencing Edited One Handler](https://github.com/guilhermejccavalcanti/jFSTMerge/blob/master/src/main/java/br/ufpe/cin/mergers/handlers/NewElementReferencingEditedOneHandler.java)
 
-  `java -jar pathto/jFSTMerge.jar -d "mine" "base" "theirs" -o "output"`
+* **Description** If a developer adds a **class member** that references an edited one inside its body by another developer, semistructured merge can change the semantic of the code since it doesn't care about the elements' bodies. Note that structured merge, on the other hand, deals with this naturally. This handler attempts to fix this issue.
+* **Behavior** After detecting such members, we report a conflict if unstructured merge also reported it.
 
-Where *mine*, *base*, *theirs* and *output* are directory paths.
-The attribute -o is optional, if omitted, *theirs* is used as the output directory.
+### [Type Ambiguity Error Handler](https://github.com/guilhermejccavalcanti/jFSTMerge/blob/master/src/main/java/br/ufpe/cin/mergers/handlers/TypeAmbiguityErrorHandler.java)
 
-<!-- 
-For integration with git type the two commands bellow:
+* **Description** Semistructured merge is unable to output conflicts in **import statements**: even if an import has been edited by two different developers, they're treated as different as their identifiers are different. This can introduce a compilation error (if two imports share a same class name) or a behavioral error (a different import than intended). This handler tries to avoid such errors.
+* **Behavior** For each pair (.\*, .\*), (.\*, .A), or (.A, .A) of imports of both contributions (A is a class), we report a conflict if unstructured merge also reported it or if they introduced a compilation error in the merge.
 
-   `git config --global merge.tool jfstmerge`
-   
-   `git config --global mergetool.jfstmerge.cmd 'java -jar pathto/jFSTMerge.jar -f \"$LOCAL\" \"$BASE\" \"$REMOTE\" -o \"$MERGED\"'`
+### [Duplicated Declaration Handler](https://github.com/guilhermejccavalcanti/jFSTMerge/blob/master/src/main/java/br/ufpe/cin/mergers/handlers/DuplicatedDeclarationHandler.java)
+* **Description** This handler is special. It only detects cases where unstructured merge would allow two methods with different signatures (just as in the example above) in the result. As semistructured merge deals with this naturally, this handler serves for statistics purposes only.
```

Comment:
tira isso "This handler is special."

example above?

diz, de forma direta, o que o handler faz. ele é chamado... ele verifica se... se for, ele contabiliza tal info escrevendo em...